### PR TITLE
Fix plugin lifecycle and host edge cases (#147)

### LINKED ISSUE #147: Engine misc: plugin slot re-publish gap, CLAP thread stamp, VST3 nits
src/engine (release/0.12)

- PluginSlot.cpp:361 the in-process prepareToPlay branch re-prepares ownedInstance but does not re-publish currentInstance, which releaseResources() nulls. The OOP twin re-publishes with a comment naming the failure. Latent today because release is shutdown-only, but any future non-terminal release/re-prepare cycle silently kills every slot.
- ClapInstance.cpp:230 deactivate() stamps the calling message thread as the audio thread and never restores it. A bypassed slot never reaches the re-stamp in process, so after any reactivate (buffer size change) isAudioThread and isMainThread both return true on the message thread indefinitely.
- Vst3Instance.cpp:169 destroy() never resets controllerIsComponent, so a reused create() landing a separate controller class skips controller terminate. Latent until create() is reused.
- Vst3Instance.cpp:56 the in/out EventList default maxSize is 50 and setMaxSize is never called; addEvent silently drops past 50 events per block, stuck notes on dense MIDI. CLAP scratch is 1024 events, LV2 8192 bytes.
- Vst3HostContext.cpp:140 poll requests POLLIN | POLLOUT | POLLERR, so revents is nonzero on every pump and onFDIsSet fires for every fd at pump rate regardless of readability. Request POLLIN only.

### PR DESCRIPTION
**Pull request by marc-korte**
## What

- Republish in-process plugin instances after release and prepare cycles.
- Restore CLAP audio-thread stamping after stop processing.
- Reset reused VST3 controller state and raise MIDI event-list capacity to 1024.
- Poll VST3 run-loop descriptors only for readable/error states and retire invalid descriptors.
- Add focused CLAP thread-check and VST3 run-loop regressions.

## Why

Lifecycle transitions could silently leave slots unpublished, CLAP could permanently classify the message thread as audio, reused VST3 controllers could skip termination, dense MIDI could exceed the default event capacity, and writable descriptors could be dispatched continuously.

## Validation

- DuskStudio app build
- 539/539 tests passed
- JUCE gate: 179 files / 9257 uses

Fixes #147

## Summary by CodeRabbit

- **Bug Fixes**
  - Prevented audio plugins from becoming silent after reconfiguration or preparation.
  - Improved CLAP plugin deactivation and audio-thread handling for more reliable processing.
  - Improved VST3 event handling, including safer recovery from closed or invalid event sources.
  - Increased the supported number of VST3 input and output events per audio block.
  - Improved plugin lifecycle cleanup and state reset behavior.

- **Tests**
  - Added coverage for CLAP thread handling and VST3 event-loop edge cases.

### CODE REVIEW

**Comment by marc-korte** [comment]
@coderabbitai review

**Comment by marc-korte** [comment]
@coderabbitai review

File: src/engine/PluginSlot.cpp [hunk 1]
```diff
@@ -372,6 +372,10 @@ void PluginSlot::prepareToPlay (double sampleRate, int blockSize)
         ownedInstance->prepareToPlay (sampleRate, preparedBlockSize);
         cachedLatencySamples.store (ownedInstance->getLatencySamples(),
                                       std::memory_order_relaxed);
+        // Re-publish the live instance last: releaseResources() nulls
+        // currentInstance while keeping ownedInstance, so without this a
+        // release / re-prepare cycle leaves the slot silent until a reload.
+        currentInstance.store (ownedInstance.get(), std::memory_order_release);
```
[the remark this answers is not in the document]
Comment (marc-korte) [reply src/engine/PluginSlot.cpp:378]: Fixed in 66756bc. The new fake in-process instance test exercises install, releaseResources, prepareToPlay, and a subsequent stereo process block, asserting the republished instance receives processing. The test seam reuses the existing instance pointer alias so the JUCE gate remains 179 files / 9257 uses.
